pageserver/src/tenant/timeline.rs

```diff
@@ -4016,11 +4016,14 @@ impl Timeline {
         ctx: &RequestContext,
     ) -> Result<((KeyPartitioning, SparseKeyPartitioning), Lsn), CompactionError> {
         let Ok(mut partitioning_guard) = self.partitioning.try_lock() else {
-            // NB: there are two callers, one is the compaction task, of which there is only one per struct Tenant and hence Timeline.
+            // NB: there are three callers, one is the compaction task, of which there is only one per struct Tenant and hence Timeline.
```

Comment:
Doesn't "caller" here refer to the caller of the `repartition` function? I don't think it refers to the places that try to lock `self.partition`.

Reply:
It's okay to document the other place where we do try_lock of `self.partitions`, but it would be nice to have a difference in wording between it and callers of repartition.